### ISSUE TITLE
Language Switching Issue with Shift+Alt Shortcut

### ISSUE THREAD

**Issue by runningpig66** [valinet/ExplorerPatcher#3945]
Hello, developers of ExplorerPatcher.

I am using two languages, Japanese and Chinese, on Windows 11. However, after enabling ExplorerPatcher, I often face an issue when switching between languages using the Shift+Alt shortcut. In Windows 10, the language switch used to work smoothly, but now, with ExplorerPatcher enabled, the language switch often fails. It usually takes multiple attempts for it to work, and sometimes the language switch causes some software to crash.
When I disable ExplorerPatcher, the issue is resolved.
Could you let me know what information or logs I can provide to help you better understand and diagnose this problem?

Thanks in advance for your help!

**Comment by Amrsatrio** [comment]
I cannot help you on this regard. Shift+Alt language switching works fine on 26100.1150 ARM64 so not sure what to fix.

**Comment by Amrsatrio** [comment]
What's your Windows build?

**Comment by wangyan-life** [comment]
As I recall, the shortcut for switching languages is WIN+space?

**Comment by Amrsatrio** [comment]
Yes. And Alt+Shift is to alternate between the current one and the last selected language/layout, you can feel it if you have 3 layouts added. Win+Space and Alt+Shift both can be turned off through deep in the Control Panel.

**Comment by runningpig66** [comment]
I will now use a GIF animation to simulate the scenario where the input method switching fails during typing. Each time I switch languages using Alt+Shift (switching between Chinese and Japanese), I will record the corresponding text in Notepad++. Theoretically, if the input method switches successfully, it should alternate between one line of Chinese and one line of Japanese. However, in reality, when the input method fails to switch, it inputs the current language again. Please pay attention to the language indicator box in the lower right corner of my desktop, where the icons for Chinese and Japanese will appear. Additionally, I will provide some ExplorerPatcher and Windows input method settings as reference.
![test](https://github.com/user-attachments/assets/31c02b2b-6df4-419f-9713-39e793249e63)
![test2](https://github.com/user-attachments/assets/0290102f-423e-401c-a0b4-b338dd1d2962)

**Comment by runningpig66** [comment]
![image](https://github.com/user-attachments/assets/69e048c7-d511-4eaa-abc2-6953c33cc6f2)
![image](https://github.com/user-attachments/assets/fe480c6b-5aba-4382-9509-672de334cd8e)
![image](https://github.com/user-attachments/assets/bd6424f9-2eea-4d5e-b629-a6848cd22b7c)
![desktop](https://github.com/user-attachments/assets/8e26cbc6-72ea-4f56-bd40-a6340c5be1d7)
![image](https://github.com/user-attachments/assets/1b6fe31b-2435-4740-aa6d-10c4fffd5aa1)
![image](https://github.com/user-attachments/assets/a32b3219-168f-443d-82ff-c7bca7472d03)
![image](https://github.com/user-attachments/assets/dec59fb4-d3c1-4e89-bfef-07cfcdaa75cc)

**Comment by tomasz1986** [comment]
I can confirm similar instabilities with the language switcher. In my case, the switcher itself also seems to just randomly disappear from the taskbar. When that happens, I have to restart Explorer in order to bring it back.

All this is occurring with ExplorerPatcher v22621.4317.67.1 under Windows 11 24H2 (Enterprise LTSC 2024).

**Comment by Amrsatrio** [comment]
I'll test this on my 22631.4460 machine and see.

**Comment by Amrsatrio** [comment]
@tomasz1986 A reproduction video along with your specific Windows build number would be nice to have.

**Comment by tomasz1986** [comment]
I would love to reproduce at will but the icon disappearance seems completely random. It has happened at least a few times so far though. I think I will have to basically turn on screen recording and keep it on full time until the bug happens 😅.

As for the Windows build number, here it is.

![image](https://github.com/user-attachments/assets/923f0477-c2f9-4af1-bab0-5c7505a10fbf)
